Add asset-transfer-api docs (#16)
papermoonio/polkadot-ecosystem-docs-draft: docs/.snippets/code/build-on-polkadot/xcm/asset-transfer-api/reference/transfer-arg-opts.md

Options for customizing the claim assets transaction. These options allow you to specify the transaction format, fee payment details, weight limits, XCM versions, and more.

??? child "Show more"

    `format` ++"T extends Format"++ 
        
    Specifies the format for returning a transaction.

    ??? child "Type `Format`"

        ```ts
        export type Format = 'payload' | 'call' | 'submittable';
        ```
    
    ---

    `paysWithFeeOrigin` ++"string"++
    

>     The Asset ID to pay fees on the current common good parachain. The defaults are as follows:
> 
>         - Polkadot AssetHub - DOT
>         - Kusama AssetHub - KSM

Reviewer: ```suggestion
    The Asset ID to pay fees on the current common good parachain. The defaults are as follows:

    - Polkadot Asset Hub - `'DOT'`
    - Kusama Asset Hub - `'KSM'`
```

If these are the string values, we should represent them as such. I also update the indentation because as it was indented it was rendering as a list in a code snippet, now it just renders as a list - weird framework stuff haha 😅

    ---

    `paysWithFeeDest` ++"string"++
    
    Asset ID to pay fees on the destination parachain.

    ---

    `weightLimit` ++"{ refTime?: string, proofSize?: string }"++
    
    Custom weight limit option. If not provided, it will default to unlimited.

    ---

    `xcmVersion` ++"number"++
    
    Sets the XCM version for message construction. If this is not present a supported version will be queried, and if there is no supported version a safe version will be queried.

    ---

    `keepAlive` ++"boolean"++
    

>     Enables `transferKeepAlive` for local asset transfers. For creating local asset transfers, if true this will allow for a `transferKeepAlive` as opposed to a `transfer`.

Reviewer: ```suggestion
    Enables `transferKeepAlive` for local asset transfers. For creating local asset transfers, if `true` this will allow for a `transferKeepAlive` as opposed to a `transfer`.
```

    ---

    `transferLiquidToken` ++"boolean"++
    

>     Declares if this will transfer liquidity tokens. Default is false.

Reviewer: ```suggestion
    Declares if this will transfer liquidity tokens. Default is `false`.
```

    ---

    `assetTransferType` ++"string"++
    
    The XCM transfer type used to transfer assets. The `AssetTransferType` type defines the possible values for this parameter.

    ??? child "Type `AssetTransferType`"

>         ```ts
>         export type AssetTransferType = LocalReserve | DestinationReserve | Teleport | RemoteReserve;
>         ```

Reviewer: ```suggestion
        ```ts
        export type AssetTransferType =
          | LocalReserve
          | DestinationReserve
          | Teleport
          | RemoteReserve;
        ```
```

        !!! note
            To use the `assetTransferType` parameter, which is a string, you should use the `AssetTransferType` type as if each of its variants are strings. For example: `assetTransferType = 'LocalReserve'`.

    ---

    `remoteReserveAssetTransferTypeLocation` ++"string"++
    
    The remove reserve location for the XCM transfer. Should be provided when specifying an `assetTransferType` of `RemoteReserve`.

    ---

    `feesTransferType` ++"string"++
    
    XCM TransferType used to pay fees for XCM transfer. The `AssetTransferType` type defines the possible values for this parameter.

    ??? child "Type `AssetTransferType`"

>         ```ts
>         export type AssetTransferType = LocalReserve | DestinationReserve | Teleport | RemoteReserve;
>         ```

Reviewer: ```suggestion
        ```ts
        export type AssetTransferType =
          | LocalReserve
          | DestinationReserve
          | Teleport
          | RemoteReserve;
        ```
```
